### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by eds832**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -1,19 +1,127 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+
+const Counter = (props) => {
```
Comment (stepan-lagunovsky) [inline src/index.tsx]: A) In such cases please use object distructuring on component level.

B) `src/components/...`
1) Please create a new folder `src/components` and inside create folder for each single component. 
2) Then import and use them here.

File: src/index.tsx [hunk 2]
```diff
@@ -1,19 +1,127 @@
...
+	const [value, setValue] = React.useState(props.initialValue);
+
+	function increment() {
```
Comment (stepan-lagunovsky) [inline src/index.tsx]: Please use arrow functions `const increment = () => {...`

File: src/index.tsx [hunk 3]
```diff
@@ -1,19 +1,127 @@
...
+		setValue(value + 1);
+	}
+
+	function decrement() {
+		setValue(value - 1);
+	}
+
+	return React.createElement(
```
Comment (stepan-lagunovsky) [inline src/index.tsx]: there is no any issue here, but it's a bit cleaner to simply use distructuring here as well. 

`import { createElement } from 'react';`

`return createElement(`

File: src/index.tsx [hunk 4]
```diff
@@ -1,19 +1,127 @@
...
+		'div',
+		null,
+		React.createElement('span', null, value),
+		React.createElement('button', { onClick: increment }, '+ ADD MOVIE'),
+		React.createElement('button', { onClick: decrement }, '-')
+	);
+};
+
+const SearchForm = (props) => {
```
Comment (stepan-lagunovsky) [inline src/index.tsx]: Keeping in mind comments above - please refactor part below.

File: src/index.tsx [hunk 5]
```diff
@@ -1,19 +1,127 @@
...
 const root = ReactDOM.createRoot(
-  document.getElementById('root') as HTMLElement
+	document.getElementById('root') as HTMLElement
 );
+
 root.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+	<div className='container'>
```
Comment (stepan-lagunovsky) [inline src/index.tsx]: 1) Please replace code in `App` with this piece. Just remove tests for now.
2) Import all the required components
3) Then simply Import `App` here like it was previously

File: src/index.tsx [hunk 6]
```diff
@@ -1,19 +1,127 @@
...
+		</header>
+		<GenreSelect
+			genres={['All', 'Documentary', 'Comedy', 'Horor', 'Crime', 'Drama']}
+			selectedGenre='All'
+			onSelect={(genre) => console.log('Selected genre:', genre)}
+		/>
+		<div className='movie-counter'>
+			<span>39 movies found</span>
+		</div>
+		<div className='movie-grid'>
```
Comment (stepan-lagunovsky) [inline src/index.tsx]: Nice place to start map items. For now the data can be some fake data:
```

const MOVIES_DATA = [
	{
		name: 'Movie 1',
		src: 'https://via.placeholder.com/300x450.png?text=Movie+2',
	},
  ...
];



	<div className='movie-grid'>
		{MOVIES_DATA.map(({ name, src }) => (
			<div key={name}>
				<img src={src} />
				<h2>{name}</h2>
			</div>
		))}
	</div>
```

File: src/components/SearchForm/SearchForm.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+
+import './SearchForm.css';
+
+export type SearchFormType = {
```
Comment (stepan-lagunovsky) [inline src/components/SearchForm/SearchForm.tsx]: Please check and remove interfaces/types if you do not use them in other places

File: src/components/SearchForm/SearchForm.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+	initialQuery: string;
+	onSearch: (query: string) => void;
+};
+
+const SearchForm = (props: SearchFormType) => {
```
Comment (stepan-lagunovsky) [inline src/components/SearchForm/SearchForm.tsx]: This is a way to describe types for functions.
For arrow functions it is better to use this approach. 

```
import { FC } from 'react';
const SearchForm: FC<SearchFormType> = (props) => {}
```
And use prop destruction as well
```
const SearchForm: FC<SearchFormType> = ({ 
 initialQuery,
 onSearch,
 ...
 }) => {}
```
And update all components this way.


BTW.
Sometimes we need to do this this way:
```
const SearchForm: FC<SearchFormType> = (props) => {
        const { initialQuery, onSearch,... } = props;
}
```
but more often you will see the first approach: `= ({...}) =>...`

File: src/index.tsx [hunk 9]
```diff
@@ -5,12 +5,12 @@ import App from './App';
...
 
 const root = ReactDOM.createRoot(
-  document.getElementById('root') as HTMLElement
+	document.getElementById('root') as HTMLElement
 );
 root.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+	<React.StrictMode>
```
Comment (stepan-lagunovsky) [inline src/index.tsx]: For developing perspective we can remove this, to avoid double logs in console.

**Review by stepan-lagunovsky — APPROVED** [review]

File: src/components/GenreSelect/GenreSelect.tsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import React, { FC } from 'react';
+
+import './GenreSelect.css';
+
+type GenreSelectType = {
```
Comment (Lultik) [inline src/components/GenreSelect/GenreSelect.tsx:5]: I guess it is better to use `interface` instead. type and interface can be used to define props in a React component, but they have some differences in functionality. interface is useful for defining simple props that can be extended or reused, while type is useful for defining more complex types in your props

**Pull request merged**